### ISSUE TITLE
`pkg.main` is always parsed as a script, even if it's not

### ISSUE THREAD

**Issue by Rich-Harris** [vercel/nft#311]
I would categorise this as a super-duper-edge case were it not for the fact that the offending package, [@mapbox/node-pre-gyp](https://www.npmjs.com/package/@mapbox/node-pre-gyp), has around 3.3m weekly downloads.

That package's [`package.json`](https://github.com/mapbox/node-pre-gyp/blob/c2a746bc67df928ea5c8bb77b59daadfe41a0a59/lib/util/nw-pre-gyp/package.json) contains the following:

```
"main": "index.html"
```

Unsurprisingly, this blows up: https://github.com/sveltejs/kit/issues/5577#issuecomment-1246078342.

Perhaps `pkg.main` should be treated as a script or asset based on the same [extension logic](https://github.com/vercel/nft/blob/ac895d73248ec55d27335b7cc65aa51f5a53fe20/src/node-file-trace.ts#L351) as other files?

**Comment by styfle** [comment]
Yeah I think we could exclude `.html` even if explicitly defined in `main`.

Technically this usage is valid today if `index.html` is actually a JS file and Node.js will happily accept it.

But in practice that is probably not common and so we could skip the parse step.

Another alternative is to parse and fail, then ignore the warning if it ends with `.html`.
